kit/transport/http/middleware_test.go

```diff
@@ -43,3 +45,41 @@ func Test_normalizePath(t *testing.T) {
 		})
 	}
 }
+
+func Test_skipCors(t *testing.T) {
+	tests := []struct {
+		name    string
+		method  string
+		headers []string
+	}{
+		{
+			name:   "No Origin",
+			method: "OPTIONS",
+		},
+		{
+			name:    "With Origin",
+			method:  "OPTIONS",
+			headers: []string{"Origin"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req, err := http.NewRequest(tt.method, "/", nil)
+
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			for _, header := range tt.headers {
+				req.Header.Set(header, "value")
+			}
+
+			rr := httptest.NewRecorder()
+			handler := http.HandlerFunc(SkipOptions(nil))
```

Comment:
likely seeing a panic in when the wrapped handler(nil) is called here. One other thing I'd recommend is checking out the `pkg/testttp` pkg here in influxdb. It makes these tests super easy to build... I may be biased b/c I wrote the pkg, but still its handy 😬:

```go
svr := kithttp.SkipOptions(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
	w.Write([]byte(tt.expectedBody))
}))

testttp.
	HTTP(t, tt.method, "/", nil).
	Do(svr).
	ExpectStatus(http.StatusOK).
	ExpectBody(func(body *bytes.Buffer) {
		assert.Equal(t, tt.expectedBody, body.String())
	})
```

I added a expected body value to make sure the interior handler wasn't called. Smalll extension to the testcase struct